### ISSUE TITLE
Test failure with libxml2 2.12.0 in Fedora Rawhide

### ISSUE THREAD

**Issue by musicinmybrain** [python-openxml/python-docx#1302]
In Fedora Rawhide, the development version of the distribution, I’ve started seeing one test failure. There were many packages updated [around the time it started](https://koschei.fedoraproject.org/build/16714526), but the one that jumps out at me as a likely cause is that `libxml2` was updated from 2.11.5 to 2.12.0.

```
=================================== FAILURES ===================================
_ DescribeParseXml.it_accepts_unicode_providing_there_is_no_encoding_declaration _

self = <tests.oxml.test__init__.DescribeParseXml object at 0x7fdd816a57c0>

    def it_accepts_unicode_providing_there_is_no_encoding_declaration(self):
        non_enc_decl = '<?xml version="1.0" standalone="yes"?>'
        enc_decl = '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>'
        xml_body = "<foo><bar>føøbår</bar></foo>"
        # unicode body by itself doesn't raise
        parse_xml(xml_body)
        # adding XML decl without encoding attr doesn't raise either
        xml_text = "%s\n%s" % (non_enc_decl, xml_body)
>       parse_xml(xml_text)

tests/oxml/test__init__.py:63:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../BUILDROOT/python-docx-1.1.0-3.fc40.x86_64/usr/lib/python3.12/site-packages/docx/oxml/parser.py:27: in parse_xml
    return cast("BaseOxmlElement", etree.fromstring(xml, oxml_parser))
src/lxml/etree.pyx:3257: in lxml.etree.fromstring
    ???
src/lxml/parser.pxi:1916: in lxml.etree._parseMemoryDocument
    ???
src/lxml/parser.pxi:1796: in lxml.etree._parseDoc
    ???
src/lxml/parser.pxi:1085: in lxml.etree._BaseParser._parseUnicodeDoc
    ???
src/lxml/parser.pxi:618: in lxml.etree._ParserContext._handleParseResultDoc
    ???
src/lxml/parser.pxi:728: in lxml.etree._handleParseResult
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E     File "<string>", line 1
E   lxml.etree.XMLSyntaxError: Blank needed here, line 1, column 21

src/lxml/parser.pxi:657: XMLSyntaxError
```

**Comment by ofaurax** [comment]
I reported it here: https://gitlab.gnome.org/GNOME/libxml2/-/issues/661

**Comment by liamdiprose** [comment]
Upstream: Fixed in 2.12.4

https://gitlab.gnome.org/GNOME/libxml2/-/issues/661#note_1971394

**Comment by musicinmybrain** [comment]
> Upstream: Fixed in 2.12.4
> 
> https://gitlab.gnome.org/GNOME/libxml2/-/issues/661#note_1971394

Thanks! Now that we have libxml2 2.12.4 in Fedora Rawhide, I can confirm that this test works again for me too.
